spinetoolbox/spine_db_editor/mvcmodels/pivot_table_models.py

```diff
@@ -929,6 +929,8 @@ def __init__(self, parent):
             accepts_item=self._parent.accepts_entity_item,
             owner=self,
         )
+        self._entity_fetch_parent.setParent(self)
+        self.destroyed.connect(lambda: self._entity_fetch_parent.set_obsolete(True))
```

Comment:
@soininen in the FetchParent constructor, if owner is an instance of QObject, we connect its destroyed signal exactly as you did here. Was that not working as expecting? What was the reason to doing it this way?

Reply:
I missed the connection in the constructor and just did a blanket fix. It did fix the problem so I assume some changes in this PR are correct :) I'll get back to the drawing board and dig deeper into this. Thanks for the heads-up!

Reply:
> Do you know by the way if the QObject.destroyed signal is supposed to have any arguments? 

It is supposed to have the `obj` argument but Qt documentation does not imply when it would be `nullptr`. I erroneously removed the argument from one of the lambdas here because I somehow remembered that `destroyed` did not have any arguments. On the other hand, slots seem to work fine in `PySide` if they have less arguments than the signal so sometimes I am tempted to remove the arguments if they are not used in the slot at all...